File: 프로그래머스/1/12977. 소수 만들기/소수 만들기.java
import java.util.*;
class Solution {
    public int solution(int[] nums) {
        int ret = 0;
        Arrays.sort(nums);
        int max = nums[nums.length - 1] + nums[nums.length - 2] + nums[nums.length - 3];
        int[] che = new int[max + 4];
        
        for(int i = 2; i <= max; i++){
            if(che[i] == 1) continue;
            for(int j = i * 2; j <= max; j += i){
                che[j] = 1;
            }
        }
        
        for(int i = 0; i < nums.length; i++){
            for(int j = 0; j < i; j++){
                for(int k = 0; k < j; k++){
                    int sum = nums[i] + nums[j] + nums[k];
                    if(che[sum] == 1) continue;
                    ret++;
                }
            }
        }
        
        
        return ret;
    }
}

File: 프로그래머스/1/12977. 소수 만들기/소수 만들기.py
def solution(nums):
    size = len(nums)
    ssums = []
    for i in range(size):
        for j in range(i+1, size):
            for k in range(j+1, size):
                ssums.append(nums[i] + nums[j] + nums[k])
    n = max(ssums)
    prime = [False] * (n+1)
    prime[0] = prime[1] = True
    for i in range(2, n+1):
        if prime[i]: continue
        for j in range(i+i, n+1, i):
            prime[j] = True
    return sum(1 for x in ssums if not prime[x])
    